mParticle-Apple-SDK/Data Model/MPConsumerInfo.mm

```diff
@@ -307,46 +322,52 @@ - (NSDictionary *)localCookiesDictionary {
 
 #pragma mark Public accessors
 - (NSNumber *)mpId {
-    if (_mpId) {
-        return _mpId;
-    }
-    
-    [self willChangeValueForKey:@"mpId"];
-    
-    NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-    NSString *mpIdString = userDefaults[kMPRemoteConfigMPIDKey];
-    
-    if (mpIdString) {
-        [userDefaults removeMPObjectForKey:kMPRemoteConfigMPIDKey];
-
-        if (sizeof(void *) == 4) { // 32-bit
-            _mpId = [NSNumber numberWithLongLong:(long long)[mpIdString longLongValue]];
-        } else if (sizeof(void *) == 8) { // 64-bit
-            _mpId = [NSNumber numberWithLong:(long)[mpIdString longLongValue]];
+    // Must ensure it's created prior to allowing it to be created
+    dispatch_sync(self.queue, ^{
+        // If we don't have the id, create it.
+        if (!_mpId) {
+            [self willChangeValueForKey:@"mpId"];
+            
+            NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+            NSString *mpIdString = userDefaults[kMPRemoteConfigMPIDKey];
+            
+            if (mpIdString) {
+                [userDefaults removeMPObjectForKey:kMPRemoteConfigMPIDKey];
```

Comment:
Can we reduce the scope of this `dispatch_sync`? The access to `userDefaults` here may need to happen on the main thread.